src/content/docs/guides/Beginners/Your First Shader/1_composite.mdx

```diff
@@ -0,0 +1,167 @@
+
+First, let's open `composite.vsh`. This is the *vertex shader* for the `composite` program. Since `composite` is a fullscreen pass, this actually just renders a singular quad to the screen which exactly covers it. This means that your fullscreen passes are technically actually running on 3D geometry!
+
+```glsl
+#version 330 compatibility
+
+out vec2 texcoord;
+
+void main() {
+	gl_Position = ftransform();
+	texcoord = (gl_TextureMatrix[0] * gl_MultiTexCoord0).xy;
+}
+```
+
+Let's analyse this.
+
+```glsl
+#version 330 compatibility
+```
+
+This tells the shader what version of GLSL to use, as covered earlier.
+
+```glsl
+out vec2 texcoord;
+```
+
+This is a variable declaration, but a special one. The `out` keyword means that the value will be passed to the fragment shader. The fragment shader can then have a corresponding `in` declaration which allows it to recieve this value. 
+
+:::note[Note]
+Values passed from vertex shaders to fragment shaders are interpolated! This means that if one vertex has a value of 0, and another has a value of 1, then a pixel halfway between these two vertices will have a value of 0.5. This can be prevented by adding the `flat` keyword the `in` and `out` declarations.
+:::
+
+```glsl
+void main() {
+```
+
+The `main` function is the code that is run when the shader is invoked, just like in languages like C.
+
+```glsl
+gl_Position = ftransform();
+```
+
+This transforms the position of the vertex from model space to clip space. The `ftransform` function is actually deprecated but Iris patches it to the relevant modern code. We will revisit this later.
+
+```glsl
+texcoord = (gl_TextureMatrix[0] * gl_MultiTexCoord0).xy;
+```
+
+This gives us the 'texture coordinate' of the current vertex. This is more commonly known as the 'UV', and it is used so that the fragment shader knows where on the screen it is. You do not need to understand how it works at this stage. These texture coordinates range from (0, 0) at the bottom left of the texture to (1, 1) at the top right.
+
+
+Let's open `composite.fsh`. This is the *fragment shader*  It runs for every pixel on the screen.
+
+Right now the code looks like this
+
+```glsl
+#version 330 compatibility
+
+uniform sampler2D colortex0;
+
+in vec2 texcoord;
+
+/* DRAWBUFFERS:0 */
+layout(location = 0) out vec4 color;
+
+void main() {
+	color = texture(colortex0, texcoord);
+}
+```
+
+Let's analyse this as well a bit.
+
+
+```glsl
+uniform sampler2D colortex0
+```
+
+This allows the shader to read from `colortex0`. Iris provides you with a number of `colortex` buffers you can read and write to. Each buffer is a texture with the same resolution as your screen. For more information on the `colortex` buffers, see the [docs](/reference/buffers/colortex/).
+
+```glsl
+in vec2 texcoord
+```
+
+This is where we recieve the `texcoord` passed `out` in the vertex shader. This tells us what pixel to sample in `colortex0`.
+
+```glsl
+/* DRAWBUFFERS:0 */
```

Comment:
Would rendertargest be better to teach?

Reply:
Yes, but the base pack uses drawbuffers and I didn't want to have the user just go and replace them. Maybe the base pack should be updated to use rendertargets?

Reply:
https://github.com/shaderLABS/Base-330/issues/3